spec/app/domain/authentication/authn-azure/validate_azure_annotations_spec.rb

```diff
@@ -0,0 +1,167 @@
+                                                             .and_return(subscription_id_annotation_service_id_scoped)
+    allow(subscription_id_annotation_service_id_scoped).to receive(:[])
+                                                             .with(:name)
+                                                             .and_return("authn-azure/#{good_service_id}/subscription-id")
+    allow(subscription_id_annotation_service_id_scoped).to receive(:[])
+                                                             .with(:value)
+                                                             .and_return("AzureSubscriptionIdServiceIdScoped")
+
+    allow(resource_group_annotation).to receive(:values)
+                                          .and_return(resource_group_annotation)
+    allow(resource_group_annotation).to receive(:[])
+                                          .with(:name)
+                                          .and_return("authn-azure/resource-group")
+    allow(resource_group_annotation).to receive(:[])
+                                          .with(:value)
+                                          .and_return("some-resource-group-value")
+
+    allow(resource_group_annotation_service_id_scoped).to receive(:values)
+                                                            .and_return(resource_group_annotation_service_id_scoped)
+    allow(resource_group_annotation_service_id_scoped).to receive(:[])
+                                                            .with(:name)
+                                                            .and_return("authn-azure/#{good_service_id}/resource-group")
+    allow(resource_group_annotation_service_id_scoped).to receive(:[])
+                                                            .with(:value)
+                                                            .and_return("AzureResourceGroupServiceIdScoped")
+
+    allow(invalid_azure_annotation).to receive(:values)
+                                   .and_return(invalid_azure_annotation)
+    allow(invalid_azure_annotation).to receive(:[])
+                                   .with(:name)
+                                   .and_return("authn-azure/non_existing")
+
+    allow(non_azure_annotation).to receive(:values)
+                                   .and_return(non_azure_annotation)
+    allow(non_azure_annotation).to receive(:[])
+                                   .with(:name)
+                                   .and_return("authn-test/non_existing")
+
+    allow(invalid_azure_annotation_service_id_scoped).to receive(:values)
+                                     .and_return(invalid_azure_annotation_service_id_scoped)
+    allow(invalid_azure_annotation_service_id_scoped).to receive(:[])
+                                                             .with(:name)
+                                                             .and_return("authn-azure/#{good_service_id}/non-existing")
```

Comment:
Can this be extracted to a method to enhance readability?

Reply:
in the same fashion as detailed under the previous extraction recommendation?

```
def mock_annotation_builder(host, host_annotation, host_value)
  allow(host).to receive(:values)
                                   .and_return(host)
  allow(host).to receive(:[])
                                   .with(:name)
                                   .and_return("host_annotation")
end

before(:each) do
  mock_annotation_builder(invalid_azure_annotation_service_id_scoped, authn-azure/#{good_service_id}/non-existing)
  mock_annotation_builder ...
 mock_annotation_builder ...
end
```

Reply:
yes, something like that.